src/lib.rs

```diff
@@ -1 +1,76 @@
+use std::{ffi::OsString, path::Path, process::Command};
 
+/// Options provided for compiling binaries.
+struct CompileOption {
+    /// File to execute when compile.
+    pub compiler: OsString,
+    /// Arguments passed to compiler process.
+    pub args: Vec<OsString>,
+}
+
+/// Result of compilation.
+struct CompileResult {
+    /// If compiler exits normally then `true`, else `false`.
+    pub success: bool,
```

Comment:
Why don't we store an exit code?

Reply:
Storing exit code makes io error be handled otherwise. I wanted to avoid this but given that it's `evalsa`-internal API, there is no reason to drop exit code support. Will change.